app/src/modules/settings/routes/data-model/utils/get-junction-collection-name.ts

```diff
@@ -0,0 +1,24 @@
+import { collectionExists } from './collection-exists';
+
+export function getAutomaticJunctionCollectionName(collectionA: string, field: string) {
+	let index = 0;
+	let name = getName(index);
+
+	while (collectionExists(name)) {
+		index++;
+		name = getName(index);
+	}
+
+	return name;
+
+	function getName(index: number) {
+		let name = `${collectionA}_${field}`;
+
+		if (name.startsWith('directus_')) {
```

Comment:
I think we can also use `isSystemCollection` here:
```suggestion
import { isSystemCollection } from '@directus/system-data';
import { collectionExists } from './collection-exists';

export function getAutomaticJunctionCollectionName(collectionA: string, field: string) {
	let index = 0;
	let name = getName(index);

	while (collectionExists(name)) {
		index++;
		name = getName(index);
	}

	return name;

	function getName(index: number) {
		let name = `${collectionA}_${field}`;

		if (isSystemCollection(name)) {
```